### ISSUE TITLE
User created by API only is created in one node.

### ISSUE THREAD

**Issue by GuiuMateu** [emqx/emqx#2201]
#### Environment

- OS: Ubuntu 16.04 LTS (4.4.0-1069-aws)
- EMQ: 2.3.11 (from http://emqtt.io/downloads/stable/ubuntu16_04-deb)

#### Description
I have a EMQTT cluster with two nodes. Both are correctly connected:
```bash
# emqttd_ctl cluster status
Cluster status: [{running_nodes,['emqtt@10.0.1.75','emqtt@10.0.1.76']}]
```

After cluster is running I want create a new user by API Rest:
```bash
curl -X POST \
  http://10.0.1.75:18083/api/v2/users \
  -H 'Authorization: Basic KSF8934JFSPODSFJ3RF9D==' \
  -H 'Cache-Control: no-cache' \
  -H 'Content-Type: application/json' \
  -d '{
	"username": "newUser",
	"password": "hisPassword",
	"tags": "user"
}'
```

This API requests is only directed to one node, the first one. **I hope that all nodes in cluster replicate this user in his systems, but it seems doesn't work.**

**Here users list for first node**
```bash
curl -X GET \
  http://10.0.1.75:18083/api/v2/users \
  -H 'Authorization: Basic KSF8934JFSPODSFJ3RF9D==' \
  -H 'Cache-Control: no-cache' \
```
```json
{
    "code": 0,
    "result": [
        {
            "username": "admin",
            "tags": "administrator"
        },
        {
            "username": "newUser",
            "tags": "user"
        }
    ]
}
```

**Here users list for second node**
```bash
curl -X GET \
  http://10.0.1.76:18083/api/v2/users \
  -H 'Authorization: Basic KSF8934JFSPODSFJ3RF9D==' \
  -H 'Cache-Control: no-cache' \
```
```json
{
    "code": 0,
    "result": [
        {
            "username": "admin",
            "tags": "administrator"
        }
    ]
}
```

Is it right? Do I should do request to each cluster node?

Thank you

**Comment by Gilbert-Wong** [comment]
In the latest implementation of emqx cluster mode, one client/session just exist in one node, it would not be backup or migrated in different nodes. We would redesign it in the future.

**Comment by GuiuMateu** [comment]
Hi @Gilbert-Wong,

I was talking about users, not clients/sessions. I agree that one MQTT client with his session only exist in one node, but I think that users (available to manage cluster by API or web) should exists in all nodes.

If I have one cluster with 3 nodes and a new admin user is created by API Rest (POST api/v2/users) in one node, the other two don't have this user, so I can't login in management web (18083) on these nodes.

Is that correct?

Thank you

**Comment by danigranados** [comment]
> In the latest implementation of emqx cluster mode, one client/session just exist in one node, it would not be backup or migrated in different nodes. We would redesign it in the future.

If users are not replicated at all the nodes, when they send a request to the API with their credentials at basic auth header they obtain a 401 when load balancer sends the request to the node that has not that user information. If there are two nodes, then you have that half of the times user get an unauthorized response at login.

**Comment by Gilbert-Wong** [comment]
Sorry for late replay, @danigranados , the authentication mechanism depends the auth plugin on the local node which the connection exist.

**Comment by HJianBo** [comment]
Hi, @GuiuMateu @danigranados 
>  but I think that users (available to manage cluster by API or web) should exists in all nodes

I agree that, and the EMQ X follow this design all the time (whatever v2.3.11 and latest version).

I'll check it tomorrow. Thanks guys to feedback these!

**Comment by HJianBo** [comment]
It is actually a bug. The admin users info of dashboard should keep sync in the cluster. I'll fix it later!

Thanks for your feedback.

**Comment by HJianBo** [comment]
Hi guys, I have fixed this issue and it will be released tomorrow

**Comment by HJianBo** [comment]
This fix has released at v3.2.rc.2. Thanks
